File: app/src/main/java/com/ericho/coupleshare/mvp/RegisterContract.java
package com.ericho.coupleshare.mvp;

import android.support.annotation.NonNull;
import android.widget.EditText;

/**
 * Created by steve_000 on 14/6/2017.
 * for project CoupleShare
 * package name com.ericho.coupleshare.mvp
 */

public interface RegisterContract {
    interface View extends BaseView<Presenter>{

        void showLoadingIndicator(boolean active);

        void showRegisterSuccess();

        void showRegisterFailure(String errorMessage);

        void showRegisterButtonState(boolean enable);

    }

    interface Presenter extends BasePresenter{

        void register(String username,String password);

        void reset();

        void assignOnTextChangeListener(@NonNull EditText usernameEditText,
                                          @NonNull EditText passwordEditText);

        void destroy();
    }
}

File: app/src/main/java/com/ericho/coupleshare/mvp/RegisterContract.kt
package com.ericho.coupleshare.mvp

import android.widget.Button
import android.widget.EditText

/**
 * Created by steve_000 on 8/7/2017.
 * for project CoupleShare
 * package name com.ericho.coupleshare.mvp
 */
interface RegisterContract {
    interface View : BaseView<Presenter> {

        fun showLoadingIndicator(active: Boolean)

        fun showRegisterSuccess()

        fun showRegisterFailure(errorMessage: String?)

        fun showRegisterButtonState(enable: Boolean)

    }

    interface Presenter : BasePresenter {

        fun register(username: String, password: String)

        fun reset()

        fun assignOnTextChangeListener(usernameEditText: EditText,
                                       passwordEditText: EditText,
                                        loginButton:Button)

        fun destroy()
    }
}